### ISSUE TITLE
Support getting polygon id when picking from surface

### ISSUE THREAD

**Issue by ahojukka5** [pyvista/pyvista#3915]
### Describe the feature you would like to be added.

I'm making a simple debugging tool in which I'm implementing the picking feature. I want it to work so that when I click on a surface with the left mouse button, I can color the selected polygon as a sign of selection.

![image](https://user-images.githubusercontent.com/1264779/215761438-b107fc8f-4ef5-4cde-9995-2dafe01eaef0.png)

I do all kinds of additional calculations based on the selection. That is why it is essential to know which polygon the selection has hit. After looking through the source code and documentation, I couldn't find any way to do this. A little modification to `enable_surface_picking`-function seems to give the necessary functionality:

```python
        def _end_pick_event(picker, event):
            # ....
            self._picked_cell_index = picker.GetCellId()
            # ...
```

with an appropriate getter. Am I right that this kind of functionality cannot be implemented with the current master version and would it be useful to implement it?

### Pseudocode or Screenshots

```python
Insert code here
```
Or drag your screenshot here!

**Comment by ahojukka5** [comment]
So basically I get the desired functionality with this modification:

```diff
diff --git a/pyvista/plotting/picking.py b/pyvista/plotting/picking.py
index b57cef3a..6803548c 100644
--- a/pyvista/plotting/picking.py
+++ b/pyvista/plotting/picking.py
@@ -488,6 +488,20 @@ class PickingHelper:
         """
         return self._picked_point

+    @property
+    def picked_cell_index(self):
+        """Return the picked cell index.
+
+        This returns the picked cell index after selecting a point with
+        :func:`<enable_surface_picking> pyvista.Plotter.enable_surface_picking`.
+
+        Returns
+        -------
+        int or None
+            Picked cell if available.
+        """
+        return self._picked_cell_index
+
     @property
     def picked_block_index(self):
         """Return the picked block index.
@@ -591,6 +605,7 @@ class PickingHelper:
         def _end_pick_event(picker, event):

             picked_point_id = picker.GetPointId()
+            self._picked_cell_index = picker.GetCellId()
             if not pickable_window and picked_point_id < 0:
                 self._picked_point = None
                 return
```

**Comment by banesullivan** [comment]
I'm going to add this to #3100

**Comment by ahojukka5** [comment]
After thinking about this a bit, my main suggestion would be to refactor the picker functionality so that it's extensible. This is the kind of functionality that user can have some non-trivial need of their own that should be able to be attached to the code. But maybe this is another story.

```python
class MyCustomPicker(AbstractPicker):
    # implementation

pl.enable_picker(MyCustomPicker, some_optional_arg=True)
```

**Comment by banesullivan** [comment]
#4254 implements a clean solution for this with the new element picking interface which will track cell IDs

**Comment by banesullivan** [comment]
After #4254, this can be done as:

```py
import pyvista as pv
from pyvista.plotting.opts import ElementType, PickerType
from pyvista import examples

mesh = examples.load_random_hills()

p = pv.Plotter()
p.add_mesh(mesh, show_edges=True, pickable=True)
p.enable_element_picking(mode=ElementType.CELL, line_width=10)
p.show()
```

```py
# Get the picked cell ID
p.picked_cell['vtkOriginalCellIds']
```
